src/fragments/lib/storage/ios/configureaccess/50_customization.mdx

```diff
@@ -0,0 +1,142 @@
+            }
+            return
+        }
+
+        getIdentityId { result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(error))
+            case .success(let identityId):
+                if accessLevel == .protected {
+                    completion(.success("myProtectedPrefix/" + identityId + "/"))
+                } else if accessLevel == .private {
+                    completion(.success("myPrivatePrefix/" + identityId + "/"))
+                }
+
+            }
+        }
+    }
+
+    public func getIdentityId(completion: @escaping (Result<String, StorageError>) -> Void) {
+        Amplify.Auth.fetchAuthSession { result in
+            do {
+                let session = try result.get()
+                if let identityProvider = session as? AuthCognitoIdentityProvider {
+                    let identityId = try identityProvider.getIdentityId().get()
+                    completion(.success(identityId))
+                }
+            } catch {
+                completion(.failure(StorageError.authError("Failed to retrieve auth session", "", error)))
+            }
+        }
+    }
+}
+```
+
+Then configure the storage plugin with the resolver.
+
+```swift
+let storagePlugin = AWSS3StoragePlugin(configuration: .prefixResolver(MyDeveloperDefinedPrefixResolver()))
+Amplify.add(storagePlugin)
+```
+
+For example, if you want to enable read, write and delete operation for all the objects under path *myPublicPrefix/*, declare it in your IAM policy:
+
+```xml
+"Statement": [
+    {
+        "Effect": "Allow",
+        "Action": [
+            "s3:GetObject",
+            "s3:PutObject",
+            "s3:DeleteObject"
+        ],
+        "Resource": [
+            "arn:aws:s3:::your-s3-bucket/myPublicPrefix/*",
+        ]
+    }
+]
+```
+
+If you want to have custom *private* path prefix like *myPrivatePrefix/*, you need to add it into your IAM policy:
+```xml
+"Statement": [
+    {
+        "Effect": "Allow",
+        "Action": [
+            "s3:GetObject",
+            "s3:PutObject",
+            "s3:DeleteObject"
+        ],
+        "Resource": [
+            "arn:aws:s3:::your-s3-bucket/myPrivatePrefix/${cognito-identity.amazonaws.com:sub}/*"
+        ]
+    }
+]
+```
+This ensures only the authenticated users has the access to the objects under the path. 
+
+#### Examples
+
+1. See [StorageAccessLevelAwarePrefixResolver](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Storage/AWSS3StoragePlugin/Configuration/AWSS3PluginPrefixResolver.swift#L26) and [StorageRequestUtils.getAccessLevelPrefix](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Storage/AWSS3StoragePlugin/Support/Utils/StorageRequestUtils+Getter.swift#L24) for an example of the prefix resolver used by the plugin. You do not have to add this yourself and is used by default if no prefix resolver is added to the configuration.
```

Comment:
I thought we are removing code snippet examples. Do we really need this point 1, these resolvers are used internally and developers do not need to know about it.

Reply:
taking a look, might have missed pushing a commit here

Reply:
ah yes, forgot to push the [commit](https://github.com/aws-amplify/docs/pull/3321/commits/631d945c7624ad8405378ae7f89c8c7b1e0fcb3b) - this removed the code snippet examples